pkg/controller/operators/olm/operator.go

```diff
@@ -904,6 +904,11 @@ func (a *Operator) transitionCSVState(in v1alpha1.ClusterServiceVersion) (out *v
 		if installErr := a.updateInstallStatus(out, installer, strategy, v1alpha1.CSVPhaseInstalling, v1alpha1.CSVReasonWaiting); installErr == nil {
 			logger.WithField("strategy", out.Spec.InstallStrategy.StrategyName).Infof("install strategy successful")
 		}
+
+		// Clean up deployments in case some become orphaned
+		if cleanupErr := a.cleanupOrphanedDeployments(logger, out); cleanupErr == nil {
```

Comment:
This should work, but I think it would make sense to call this over in the deployment installer:

https://github.com/operator-framework/operator-lifecycle-manager/blob/176bf33dd455d5fb18bad3e53d1ae45853c2e329/pkg/controller/install/deployment.go#L104-L121

And looking at this now I think it would be reasonable to:

- skip parsing the previous install strategy like it currently does
- instead of what `cleanupPrevious` is currently doing, let's switch it to what you're doing in `cleanupOrphanedDeployments`
- (as a followup in the future) give the deployment installer access to the deployment cache so we don't have to reach back to the cluster for that